### PR TITLE
Update scrapy & dependencies

### PR DESCRIPTION
**Pull request by teh**
###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by domenkozar** [comment]
cc @jskrablin

**Review by FRidh — APPROVED** [review]
haven't tested it but it looks good

**Comment by teh** [comment]
@drewkett you ok with these changes?

**Comment by drewkett** [comment]
Looks fine to me, though I'd probably add a comment to the nix file explaining why the permissions patch is needed rather than having that information only be in a commit message.

I'm not using this package at the moment, so feel free to take over as maintainer if you'd like.

**Comment by teh** [comment]
I left you in as the maintainer for now - I'm not directly using them either, had some unrelated code that needed a newer scrapy. 

Updated according to your comment.

**Comment by teh** [comment]
@FRidh - I think this is ready to merge.

File: pkgs/development/python-modules/scrapy/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{ self, buildPythonPackage, pkgs, lib }:
```
Comment (FRidh) [inline pkgs/development/python-modules/scrapy/default.nix]: - never pass in `pkgs`
- python libraries are passed the individual python packages, not pythonPackages or self

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/scrapy/default.nix (hunk 1 above)
Comment (teh) [inline pkgs/development/python-modules/scrapy/default.nix]: done, PTAL

**Comment by FRidh** [comment]
thanks

**Pull request merged**